### PR TITLE
Mobile support

### PR DESCRIPTION
**Pull request by thesauri**
Addresses #17.

- Touching with one finger draws
- Pinching with two fingers pans the screen

Known issues:
- The tooltip is tiny when zoomed out
- The crossword starts zoomed at a neutral level, which results in very little crossword content being displayed on small screens. Ideally, the crossword would start zoomed out to a level where the whole crossword is visible. However, as this only is achievable by browser zooming at the moment the tooltip would mess up by doing this. To achieve proper zooming that doesn't affect DOM elements I suspect we would had to implement panning and zooming using transformation matrices on the canvas instead, instead of relying on browser zooming and scrolling for panning.

### CODE REVIEW

**Comment by thesauri** [comment]
TypeScriptified and ready for review

**Comment by jhnj** [comment]
Taking a look at this now. As we discussed, we should add scaling for the tooltip in a later PR to fix the first of the mentioned issues. Scaling/translating canvas is slower and would require more work.

File: frontend/src/Crossword.tsx [hunk 1]
```diff
@@ -364,10 +383,41 @@ const Crossword: React.FC<CrosswordProps> = (props) => {
...
+    };
+
     window.onkeydown = handleKey;
-    canvas.onmousedown = (event) => startDrawing(event);
-    canvas.onmousemove = draw;
-    canvas.onmouseup = stopDrawing;
+    canvas.onmousedown = handleMouseDown;
+    canvas.onmousemove = handleMouseMove;
+    canvas.onmouseup = handleMouseUp;
+    canvas.ontouchstart = handleTouchStart;
```
Comment (Pingviinituutti) [inline frontend/src/Crossword.tsx:418]: Implementing the touch stuff like this got the touch drawing working for me :)
```
canvas.addEventListener("touchstart", handleTouchStart, false)
```

Comment (jhnj) [reply frontend/src/Crossword.tsx:418]: Works fine for me as is

Comment (Pingviinituutti) [reply frontend/src/Crossword.tsx:418]: Also adding this CSS property to the drawing layer fixes panning and zooming.
```
touch-action: pinch-zoom;
```
(https://css-tricks.com/almanac/properties/t/touch-action/#article-header-id-1)

Comment (Pingviinituutti) [reply frontend/src/Crossword.tsx:418]: > Works fine for me as is

RIP HP EliteBook touchscreen :D

**Comment by Pingviinituutti** [comment]
@thesauri working commit here https://github.com/Pingviinituutti/korsord/commit/62db66b8d9a37d4171dc0ac59673a3f59168fc00

**Comment by Pingviinituutti** [comment]
> LGTM!
> 
> As mentioned, we should add scaling for the tooltip + some way of detecting mobile users and then autozooming/minimizing tooltip... later.
> 
> But for now this works fine as a mobile mvp on my nokia 6.1
> 
> Although if it doesn't work correctly for @Pingviinituutti we should look into that before merging.
> 
> And lastly, a question @thesauri @Pingviinituutti : How are you running prettier? For me it changes this:
> 
> ```
>       {
>         // @ts-ignore
>         readyState === WebSocket.CLOSED && <ConnectionErrorPopup />
>       }
> ```
> 
> to this:
> 
> ```
>       {// @ts-ignore
>         readyState === WebSocket.CLOSED && <ConnectionErrorPopup />}
> ```
> 
> which typescript complains about



> LGTM!
> 
> As mentioned, we should add scaling for the tooltip + some way of detecting mobile users and then autozooming/minimizing tooltip... later.
> 
> But for now this works fine as a mobile mvp on my nokia 6.1
> 
> Although if it doesn't work correctly for @Pingviinituutti we should look into that before merging.
> 
> And lastly, a question @thesauri @Pingviinituutti : How are you running prettier? For me it changes this:
> 
> ```
>       {
>         // @ts-ignore
>         readyState === WebSocket.CLOSED && <ConnectionErrorPopup />
>       }
> ```
> 
> to this:
> 
> ```
>       {// @ts-ignore
>         readyState === WebSocket.CLOSED && <ConnectionErrorPopup />}
> ```
> 
> which typescript complains about

My prettier does not do that to me 🤔 Of course, the error could be fixed also :D

File: frontend/src/Crossword.tsx (hunk 1 above)
Comment (thesauri) [reply frontend/src/Crossword.tsx:418]: Ah, I didn’t know about that CSS property! I think that may be exactly what we’re looking for here. I’ll try that out tomorrow.

Comment (thesauri) [reply frontend/src/Crossword.tsx:418]: What browser were you running on the EliteBook?

**Comment by thesauri** [comment]
I’m running prettier in VSCode with autofix on save using `.prettierrc`

**Comment by thesauri** [comment]
I’ll merge this now. From what I understand the committed version of the prettier inconstancy discussed above seems to be correct. The CSS addition looks like exactly what we need, but I can’t test it right now and the current version works so I’ll go ahead and merge it to get it out of the way, adding the CSS fix separately.

**Comment by jhnj** [comment]
@thesauri the committed code in the prettier thing is correct, what my prettier changes it to fails typechecking...

**Pull request merged**